cli/doc/tests.rs

```diff
@@ -41,886 +42,267 @@ impl DocFileLoader for TestLoader {
   }
 }
 
-#[tokio::test]
-async fn export_fn() {
-  let source_code = r#"/**
-* @module foo
-*/
-
-/**
-* Hello there, this is a multiline JSdoc.
-*
-* It has many lines
-*
-* Or not that many?
-*/
-export function foo(a: string, b?: number, cb: (...cbArgs: unknown[]) => void, ...args: unknown[]): void {
-    /**
-     * @todo document all the things.
-     */
-    console.log("Hello world");
+macro_rules! doc_test {
+  ( $name:ident, $source:expr; $block:block ) => {
+    doc_test!($name, $source, false, false; $block);
+  };
+
+  ( $name:ident, $source:expr, details; $block:block ) => {
+    doc_test!($name, $source, true, false; $block);
+  };
```

Comment:
As a rule of thumb, in deno codebase, we try to refrain from using macros. Those helpers you created are super useful, but I'd prefer to have them as a functions with fixed number of arguments. @ry WDYT?

Reply:
They are used only in tests so I guess it's fine after all...